### ISSUE TITLE
Config option for custom directory

### ISSUE THREAD

**Issue by MTecknology** [go-gitea/gitea#2228]
It's possible to use a GITEA_CUSTOM environment variable to define where the custom/ directory is located, but managing an environment variable to run a daemon. It would be nice if this could also be a run-time and configuration option.

**Comment by mckaygerhard** [comment]
i want this feature/need too and this doe snot repect the unix standar layout for installing property...

**Comment by mckaygerhard** [comment]
this iisue was solved? why marked stalled?

**Comment by zeripath** [comment]
OK, I think just adding a --custom-dir="" option to the commands isn't really worth it. Especially as the cheapest way to implement that would be just to do `os.Setenv("GITEA_CUSTOM", providedCustomDir)`.

We need to look again what `$GITEA_CUSTOM` really represents. It provides the basis of the following paths: (I will use `$GITEA_CUSTOM` for `setting.CustomPath` here)

* `$GITEA_CUSTOM/options` the base directory of local "gitignore", "license", "readme", "label" files.
* `$GITEA_CUSTOM/public` a directory of custom public assets
* `$GITEA_CUSTOM/templates` a directory of custom templates
* `$GITEA_CUSTOM/robots.txt` the robots.txt  
* `$GITEA_CUSTOM/conf/app.ini` the default location of the app.ini <- the only separately configurable one of these

Is it acceptable for all of these things to be under say `/etc/gitea/`? (with the exception of `app.ini` being maybe pointed to `/etc/gitea.ini`) 

What I suspect is really wanted to provide a build option that would change the default of `$GITEA_CUSTOM` and `app.ini` then you don't need this new option or `-c` making `/usr/bin/gitea` just work - however, in terms of the necessity of providing this, would it be possible to explain why you can't just make the gitea binary be shadowed by a sh script that adjusts the environment like many programs which don't necessarily obey the FHS e.g. Firefox or chrome? So the original gitea binary ends up as `/usr/lib/gitea/bin/gitea`  and the shell script is `/usr/bin/gitea` with it setting the `GITEA_CUSTOM` and passing `-c ` as appropriate.

**Comment by mckaygerhard** [comment]
El sáb., 13 de abr. de 2019 a la(s) 13:31, zeripath (
notifications@github.com) escribió:

> What I suspect is really wanted to provide a build option that would
> change the default of $GITEA_CUSTOM and app.ini then you don't need this
> new optionor -c making /usr/bin/gitea just work - however, in terms of
> the necessity of providing this, would it be possible to explain why you
> can't just make the gitea binary be shadowed by a sh script that adjusts
> the environment like many programs which don't necessarily obey the FHS
> e.g. Firefox or chrome? So the original gitea binary ends up as
> /usr/lib/gitea/bin/gitea and the shell script is /usr/bin/gitea with it
> setting the GITEA_CUSTOM and passing -c as appropriate.


That its the idea! thanks for your explanation! as i always said why not
center effors to real needs like this?

**Comment by zeripath** [comment]
But why can't you use a script to shadow the Gitea binary like is done for Firefox or Chrome?

**Comment by mckaygerhard** [comment]
it's not stanar, firefox and chrome does are not installed on that way in
debian systems.. quality and organization are primary goal
Lenz McKAY Gerardo (PICCORO)
http://qgqlochekone.blogspot.com

**Comment by mckaygerhard** [comment]
1. distributions does not use those horrendous scripts.. set right paths in filesystems
2. it's not stanar, firefox and chrome does are not installed on that way in debian systems.. by example
3. a script produces issues due confuses gitea paths

simple! reasons!

**Comment by zeripath** [comment]
I'm fairly certain that /usr/bin/firefox is a script on Debian but I'm not on my box right now to prove it.

Anyway, the best we can achieve is a LDFLAGS on go install approach `LDFLAGS="-X modules/setting.CustomPath=/new/default"` this would require a simple change here:

https://github.com/go-gitea/gitea/blob/783cd649276c472aa3af97dd311eb4766ff3adfb/modules/setting/setting.go#L479-L484

So that it would check if GITEA_CUSTOM is set before trying to use it. E.g. replace L479 with:

```go
giteaCustom, ok := os.LookupEnv("GITEA_CUSTOM")
if ok {
     CustomPath = giteaCustom
}
```

In terms of changing the default app ini location (aka the -c option) that might be a little more involved but should also be doable. Similarly for some of the other environment variables.

We'd have to come up with a way of explaining why custom path is different so that users can work out why things aren't changing.

But this kind of behaviour would make the docker build a little easier overall.

Hmm... @techknowlogick what do you think?

**Comment by zeripath** [comment]
I've put the above changes in to #6631 which if merged means we can close this issue as not-needed.

**Comment by mckaygerhard** [comment]
thanks @zeripath and, umm about firefox:

```
ls -l /usr/bin/firefox-esr 
lrwxrwxrwx 1 root root 30 dic 13 06:34 /usr/bin/firefox-esr -> ../lib/firefox-esr/firefox-esr
```

**Comment by zeripath** [comment]
I'm not going to get into a cat match with you but `/usr/bin/firefox` and `/usr/bin/google-chrome` are both scripts on my system.

**Comment by mckaygerhard** [comment]
revised and confirmed, but seems i just to parse compatibility:

/usr/bin/firefox its a scritp to property use /usr/bin/firefox-esr

BUT google chrome its not a package from debian... so marked as non-valid ;-)

**Comment by zeripath** [comment]
Fixed by#6631
